### ISSUE TITLE
Incomplete screenshots for large pages (6000x6000px or more)

### ISSUE THREAD

**Issue by mstipetic** [puppeteer/puppeteer#477]
when I try to take a screenshot of a simple website, but in a large resolution the bottom part of the website doesn't look like it's being drawn.
You can see an example of a 6000x6000px website here: ![output](https://user-images.githubusercontent.com/243098/29582903-b1056b62-877e-11e7-8885-aeb56cf240f2.png)

I've tried loading contents from disk, from a remote server, using the setContent function, going slowMo, waiting on events, focusing on different elements, disable-gpu, headless true and false (one time it succeeded properly with headless:false) etc, still with the same result (which gets worse the higher the resolution_

```js
        const browser = await puppeteer.launch({slowMo: 500, headless: false, args: ['--disable-gpu', '--cast-initial-screen-width=6000', '--cast-initial-screen-height=6000']});
	const page = await browser.newPage();
	await page.setViewport({width: 6000, height: 6000})
	//await page.goto('file:///index_test.html', {waitUntil: 'networkidle'});
	await page.goto('http:///localhost:8000/index_test.html', {waitUntil: 'load'});
  //await page.setContent(template)
	//await buffer = page.screenshot({path: 'example.jpg', type: 'jpeg', quality: 100, fullPage: true});
	const buffer = await page.screenshot({type: 'png'});
	console.log('length:', buffer.length);
	await fs.writeFile('output.png', buffer);

	browser.close();
```

**Comment by Garbee** [comment]
Duplicate of #359

**Comment by mstipetic** [comment]
@Garbee are you sure it's a duplicate? I've looked at that issue before and it seems the problem there is that chrome has a hard limit on 16384px, I'm having problems on much lower resolutions than that

**Comment by samwillis** [comment]
I have found this as well, you may be able to get round it by stitching multiple screenshots taken with the 'clip' options. It's what I was going to try before coming up against a speed issue.

**Comment by vsemozhetbyt** [comment]
I recall having a similar problem with Electron. Could it be that the painting is not complete at the time of getting a screenshot? Does a timeout help?

**Comment by mstipetic** [comment]
@vsemozhetbyt maybe it's because the page is so large it captures it between drawing several frames, I don't know. I've tried timeouts and didn't help. My other problem also is that once using headless: false option it did work properly (but only once), I'm not sure what to make of that

**Comment by samwillis** [comment]
Are you on a Mac and if so have you tried it on Linux?

**Comment by mstipetic** [comment]
@samwillis I'm on a mac. Does it perform better on linux, because I'd be running it in that environment anyways?

**Comment by aslushnikov** [comment]
```js
await fs.writeFile('output.png', buffer);
```
This doesn't actually wait for the file to be written. you should use `fs.writeFileSync`.

**Comment by mstipetic** [comment]
@aslushnikov this doesn't resolve the situation, I've just tried it. Besides, the same behaviour occurs when using the 'path' argument in the screenshot function unfortunately

**Comment by aslushnikov** [comment]
@mstipetic can you provide the whole script you run? I cannot reproduce this issue otherwise.

**Comment by samwillis** [comment]
@aslushnikov I can replicate the problem this with this code:

```
const puppeteer = require('puppeteer');
(async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.setViewport({
        width: 7000,
        height: 7000
    })
    await page.goto('about:blank');
    await page.setContent('<div style="position: absolute; top: 0; left: 0; right: 0; bottom: 0; background: linear-gradient(to bottom, #b8e1fc 0%,#a9d2f3 10%,#90bae4 25%,#90bcea 37%,#90bff0 50%,#6ba8e5 51%,#a2daf5 83%,#bdf3fd 100%);"></div>')
    const start = Date.now();
    await page.screenshot({path: 'example.png'});
    console.log((Date.now()-start)+'ms')
    await browser.close();
})();
``` 

Interestingly if you remove the gradient from the div background, and give it a flat colour the image saves correctly with no missing areas.

I have tried it on both OSX and Ubuntu, both have the problem, saved image on osx:
![example-osx](https://user-images.githubusercontent.com/31130/29859026-20535c98-8d58-11e7-9952-262773f08d9c.png)

and ubuntu:
![example-ubuntu](https://user-images.githubusercontent.com/31130/29859027-20718f56-8d58-11e7-9d58-06bc70277c9c.png)

One interesting thing to note is the png seems to be being written in different size chunks on osx and ubuntu, although from a cursory glance it looks like it may be the same number of pixels that have been written...

**Comment by samwillis** [comment]
Right, I have been playing a little more with this. The code below works (red image with about half of it covered with text) however if you remove `transform: translate3d(0,0,0);` or increase the font size to 150px then it stops working.

Knowing **very** little about how chrome works, is it composited from multiple textures? could the `translate3d(0,0,0)` be forcing it to composite in a certain way? Does chrome have some sort of maximum texture size internally?

```
const puppeteer = require('puppeteer');
(async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.setViewport({
        width: 7000,
        height: 7000
    })
    await page.goto('about:blank');
    await page.setContent(`
        <body>
            <div style="transform: translate3d(0,0,0); position: absolute; top: 0; left: 0; right: 0; bottom: 0; background: #f00; font-size: 140px;">
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque ultricies ut tortor vitae fermentum. Etiam vulputate diam et semper condimentum. Morbi lacinia, metus at pellentesque volutpat, sapien diam lacinia enim, non accumsan leo purus vel purus. Vestibulum aliquam eget justo sit amet sollicitudin. Morbi hendrerit id ex ac fringilla. Nulla convallis ipsum non sapien facilisis, id venenatis lectus consequat. Maecenas lacinia convallis accumsan. Aliquam id congue tortor. Vestibulum porta iaculis massa, eget congue nibh pulvinar et. Nullam in faucibus quam, eu elementum nulla.
                Nulla odio mauris, semper eu ante ac, dictum consectetur nunc. Nulla condimentum metus est, ut feugiat eros accumsan a. Aenean sagittis risus in ipsum rutrum, in tristique augue consequat. Cras sagittis ante non est consectetur tempor. In eu pretium diam, nec aliquet nisi. In ut pulvinar elit, in fermentum justo. Praesent fermentum ante in sem suscipit, et lobortis sapien porta. Ut interdum egestas arcu, sed facilisis mi. Donec elementum nibh non nunc consequat, sed tempor mi finibus. Duis pharetra nec tortor id porttitor.
                Phasellus egestas scelerisque nunc ac eleifend. Curabitur tempus rhoncus sem, id ornare nisl pellentesque auctor. Fusce ac laoreet est. Vivamus in consectetur enim. Vivamus pharetra dignissim nibh, sed porta eros imperdiet in. Fusce cursus ipsum vitae vehicula suscipit. Nulla condimentum, felis quis maximus euismod, sem elit tristique mi, non sollicitudin sapien tortor sed elit. Aliquam non iaculis quam. Suspendisse velit massa, tempus eu fermentum id, elementum volutpat metus. Nulla mattis odio eu massa convallis sodales eu eu erat. Mauris non diam ornare urna elementum molestie id vitae nibh. Maecenas purus elit, interdum eget purus sed, viverra pharetra nunc. Curabitur non elit fermentum quam ultrices dignissim. Curabitur eu nisl id enim auctor varius non eget odio.
                Nullam venenatis pellentesque nulla, ut maximus odio lobortis non. Integer a blandit justo. Etiam quis eros ipsum. Nam et risus nec nisl consequat rutrum nec eget diam. Curabitur a ipsum efficitur felis commodo sollicitudin. Aenean efficitur finibus nisl id pellentesque. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Maecenas pretium nec metus ac sollicitudin. Suspendisse potenti. Vestibulum imperdiet massa et quam rutrum pulvinar. Pellentesque fermentum fringilla justo, quis rutrum ligula. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Mauris eleifend tincidunt placerat. Integer tincidunt tempus congue. Sed elementum eleifend velit id posuere. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Cras turpis justo, consequat ut nunc vel, rutrum convallis ex. Duis quis bibendum lacus. Sed vitae euismod lorem. Aenean fermentum elit non leo mattis, id varius mi porta. Sed semper suscipit condimentum. Vivamus in efficitur lacus, eget rhoncus dui.
            </div>
        </body>
    `)
    const start = Date.now();
    await page.screenshot({path: 'example.jpg'});
    console.log((Date.now()-start)+'ms')
    await browser.close();
})();
```

**Comment by samwillis** [comment]
I have filed a bug for this on the Chromium tracker here: [760596](https://bugs.chromium.org/p/chromium/issues/detail?id=760596)

**Comment by jeffjacobsen** [comment]
Any news on this?  I'm having a similar problem when trying to convert svg to large png images.

I've had success generating 6000x6000 px images, but it fails at 9375x7875.  The svg displays fine in Chrome browser at that size, and Phantomjs can process it.  I'm running on Ubuntu.  Increasing RAM doesn't help.  It did get a little bit further going from 2GB to 4GB.  8GB was the same as 4GB.

**Comment by OmarAbdElNasser** [comment]
Is there any solution for that problem, Please ?

**Comment by krzkaczor** [comment]
@OmarAbdElNasser @jeffjacobsen I have been working on a solution for these problems: https://github.com/VisualCop/fullscreenshooter This makes multiple screenshots and stitches them together, makes sure to crop them first etc. It works with `puppeter`. For now, works only for long vertical pages but can be adjusted to support horizontal as well.

**Comment by OmarAbdElNasser** [comment]
@krzkaczor but there will be no image for the whitespace ?

**Comment by OmarAbdElNasser** [comment]
This is my code 

const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.goto('D:\Image_2.HTML', {waitUntil: 'load'});
  function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}
console.log(page.frames())
  await page.screenshot({path: 'example.png',fullPage : true});

  await browser.close();
})();

I made so many  tests to find out why this happen.
In my case, the load event occur before finishing rendering the whole image, so if we can change waitUntil: 'load' to something like waitUntil: 'FinishRendering', it will solve the problem and i'm sure of this , the problem is that there's no option called FinishRendering, so if there's something detect the FinishingRendering instead of load, it will be great.

**Comment by krzkaczor** [comment]
@OmarAbdElNasser your method (using `fullPage`) only works reliably for small websites. If it's bigger then there are some weird artifacts etc. That's why I have created that library. 
For now, it doesn't support proper checks for "did page really loaded" but you can put timeout as you did in your snippet.

**Comment by OmarAbdElNasser** [comment]
Solved my problem using another packages , and used the logic of @krzkaczor .
https://stackoverflow.com/questions/47564602/navigate-page-till-rendering-last-frame-not-networkidel/47577503#47577503

**Comment by zth** [comment]
I'm also experiencing the same issue. I'm trying to export rather large (16k x 11k) PNG:s from an SVG, and going above like 5k in width/height starts bugging, with a majority of the content being white and not rendered.

I don't know if it helps, but it works fine using PhantomJS instead of Puppeteer (although the generated screenshot is huge, almost 800mb, which makes it unusable for me).

Can I do anything to help debugging this issue? Thanks!

**Comment by OmarAbdElNasser** [comment]
The issue is in writing the image, so please if you can do something please do it.

**Comment by ivannluz** [comment]
I have a question of puppeteer how can set the browser in order to display a full page , because if a set the size of the screen  does not matter how big is the screen always display just a part of the page like this 
![image](https://user-images.githubusercontent.com/31895571/35283989-40a54f9e-001f-11e8-91cf-57a0a4fa9564.png)

I don't want the big blank rectangle.

**Comment by ashirai** [comment]
@ivannluz, I have exactly the same problem. I can set viewport size in px but not the browser size.
Ideally I just want to set browser size in % (like width:100%, height: 100%) then the viewport changes the size depending on the specified browser size. It seems to be such a basic feature, but I can't seem to find a way to do it.

**Comment by mityok** [comment]
i'm having the same tearing issue on much smaller scale (1920x1080).
was able to recreate the issue by using @samwillis examples and increasing the number of layers:
![example](https://user-images.githubusercontent.com/2138624/39422736-505a9f80-4c77-11e8-91bf-25fea954337c.png)

`const puppeteer = require('puppeteer');
(async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.setViewport({
        width: 1920,
        height: 1080
    })
    await page.goto('about:blank');
    await page.setContent('<body>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 0; right: 0; bottom: 0; background: linear-gradient(to bottom, #b8e1fc 0%,#bdf3fd 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 1%; right: 1%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 2%; right: 2%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 3%; right: 3%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 4%; right: 4%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 5%; right: 5%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 6%; right: 6%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 7%; right: 7%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 8%; right: 8%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 10%; right: 10%; bottom: 0; background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 15%; right: 15%; bottom: 0; background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 20%; right: 20%; bottom: 0; background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 25%; right: 25%; bottom: 0; background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 30%; right: 30%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 35%; right: 35%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 40%; right: 40%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 45%; right: 45%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 0; right: 0; bottom: 0; opacity:0.8;background: rgba(255,255,255,0.8)"></div>
        </body>')
    const start = Date.now();
    await page.screenshot({path: 'example.png'});
    console.log((Date.now()-start)+'ms')
    await browser.close();
})();`

**Comment by zth** [comment]
For anyone who's interested: My current workaround for this issue is to screenshot tiles of the viewport (about 800x800, as many as I need to cover the entire viewport) to individual PNGs, and then stitch them together using Imagemagick. 

It's definitively not as easy as just taking a single large screenshot, but it works fine for my purposes. Tiles that small has worked fine so far, and I haven't had any rendering issues with this method.

**Comment by mityok** [comment]
@zth that's the direction that could work for some, unfortunately I'm taking screenshot of page with dynamic content and could have hundreds of elements, and the issue easily reproduced on small resolution of 800X600 with only 100 layers:
![example](https://user-images.githubusercontent.com/2138624/39425433-1e24d3ac-4c84-11e8-9f56-8ea278180569.png)
`const puppeteer = require('puppeteer');
(async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.setViewport({width: 800, height: 600})
    await page.goto('about:blank');
    await page.setContent(“
        <body>
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 0; right: 0; bottom: 0; background: linear-gradient(to bottom, #b8e1fc 0%,#bdf3fd 100%);opacity:0.2;"></div>
            ${Array.from(Array(100)).reduce((acc,n,i)=>“${acc}<div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: ${i/2}%; right: ${i/2}%; bottom: 0;  background: linear-gradient(to bottom, #f00 0%,#00f 100%);opacity:0.2;"></div>“,'')}
            <div style="transform: translate3d(0,0,0);position: absolute; top: 0; left: 0; right: 0; bottom: 0; opacity:0.8;background: rgba(255,255,255,0.8)"></div>
        </body>
    “)
    const start = Date.now();
    await page.screenshot({path: 'example.png'});
    console.log((Date.now()-start)+'ms')
    await browser.close();
})();`

So i'm in need of a more robust solution, perhaps there is some arguments in chrome that could address the issue?

**Comment by noschang** [comment]
People, I had the same problem when using chrome headless. It seems that the chrome renderer has issues dealing with large screenshots.

As pointed out by @zth and @mityok , the solution I found was to define a viewport size and then manually scroll the page and take a screenshot of the current visible area.

[This file](https://github.com/GoogleChrome/puppeteer/files/2658278/capture.zip) contains a script that I wrote and does exactly this.

It can be invoked from command line like this:
```node . "https://en.wikipedia.org/wiki/General_relativity"  $HOME/page-screenshot.png 1920x1080 10000 10000```

Note that for images with height above 16000px, it's necessary to adjust the "height" configuration in ```/etc/ImageMagick-6/policy.xml``` file. Also it may be necessary to adjust memory configurations depending on the page size.

The script works for most pages, but may fail when the page has dynamic content loaded on scroll event or when the page content is wrapped inside a div with overflow scroll.

**Comment by noschang** [comment]
Here is an updated version of my script.
I improved the code and removed unnecessary delay to make it faster.

[Download script](https://github.com/GoogleChrome/puppeteer/files/2680099/capture.zip)

The command line changed. Now it must be invoked this way:
```
node . --url https://en.wikipedia.org/wiki/General_relativity --viewport-width 1750 --viewport-height 1080 --output-file  $HOME/page-screenshot.png --timeout 30000 --verbose
```

**Comment by adamchenwei** [comment]
@noschang actually his solution here works perfectly, without custom script. really nice, check it out! https://github.com/GoogleChrome/puppeteer/issues/359#issuecomment-323203700

**Comment by noschang** [comment]
> @noschang actually his solution here works perfectly, without custom script. really nice, check it out! [#359 (comment)](https://github.com/GoogleChrome/puppeteer/issues/359#issuecomment-323203700)

Nice! It seems better than my solution.
I may try it out at some point.

**Comment by SarthakSri98** [comment]
@noschang , hey I tried your approach in my code but due to clip the screenshot becomes white after taking several screenshots like this. However if I just omit the clip section in my code it works fine till the last page.
I am trying to screenshot this website : https://www.postman.com/

         
           `
          //windowHeight is height of window 
          // height is total height of the site 
          while(count<10){
           let screenshotHeight = windowHeight;
            if (rsh < windowHeight) {
              screenshotHeight = windowHeight - (height - (count * windowHeight) );
            }
            buffers.push(await page.screenshot({ path:'./ex'+count+'.png', fullPage:false,clip : { 
                x: 0, 
                y: count * windowHeight, 
                width, 
                height: screenshotHeight
            }}));
}`

![685_2_a089294983670a3afdef30ddd93df98f](https://user-images.githubusercontent.com/28146914/81807942-9f374f80-953c-11ea-8204-316ffe0ca983.png)
Can you tell what could be the issue?

**Comment by noschang** [comment]
@SarthakSri98, I remember that I had similar issues. I'll give you a little explanations so you can better understand how to (possibly) solve it.

When you don't use the clip property, the Chrome engine will try to take the screenshot of the entire page at once. Depending on the height and/or width of the page this can be very memory consuming. Remember that when an image is in memory it is generally uncompressed, meaning that you have to store all the data about every single pixel (this is called a raster) plus auxiliar structures like palletes and other stuff.

Now, this is just a guess, but I believe that NodeJS or Chrome engine has some kind of memory buffer or stack that overflows in such memory demanding situations, thus causing the issues on the final image.

This is the reason why I've implemented the script using the clip property. When you use the clip property you're instructing Chrome to take a slice of the page. So, instead of dealing with a single huge image you're dealing with smaller images, one at a time.

The problem with your code is that apparently, as soon as you take the screenshots, you are pushing all your slices to a memory buffer. I believe that you're doing this because you want to join all your slices in a single image. But when you do this using buffers you create the same problem as before, because you're allocating all this data on the memory at once. I've tried this approach before and failed.

Here is the trick: instead of joining all the images in memory, you have to join them directly on disk. And this is why I use the "gm" library. This library uses a external program called "ImageMagick" that is optimized to work with large images using a disk cache. 

You can find more information here: https://imagemagick.org/script/architecture.php#tera-pixel

So, if you use the "gm" library and edit the ImageMagick config file to your needs, I believe the code will work. 

I've tested with my script and could take the screenshot of your page correctly. Also I've taken a very large screenshot without problems, watch the video below. If you have problems with the video try using the VLC player:

[capture-sample.zip](https://github.com/puppeteer/puppeteer/files/4630684/capture-sample.zip)

You can test for yourself. Follow these steps:

- Download the script from here: https://github.com/GoogleChrome/puppeteer/files/2680099/capture.zip

- Install dependencies with: ```cd capture && npm install```

- Take the screenshot using the following command line: ```node . --url https://en.wikipedia.org/wiki/Unit_colour_patch --viewport-width 1750 --viewport-height 1080 --output-file colour-patch.png --timeout 30000 --verbose```

Note that, for this to work you have to set the appropriate values on ImageMagick config file ```/etc/ImageMagick-6/policy.xml```. My config file looks like this:

```xml
  <policy domain="resource" name="memory" value="1GiB"/> <!-- Sets the maximum amount of RAM allowed. If the program overflow this quantity it starts using the disk cache -->
  <policy domain="resource" name="map" value="1GiB"/>
  <policy domain="resource" name="width" value="64MP"/> <!-- The maximum width in pixels allowed for an image. 1KP = 1.000 pixels, 1MP = 1.000.000 pixels, 1GP = 1.000.000.000 and so on -->
  <policy domain="resource" name="height" value="64MP"/> <!-- The same as above, but for height -->
  <policy domain="resource" name="area" value="1GB"/>
  <policy domain="resource" name="disk" value="50GiB"/> <!-- This is the maximum size allowed for the disk cache. This is the most important configuration if you want to join large images -->
```

**Comment by jrandolf-google** [comment]
Close due to staleness. Please reopen if the issue remains in the latest version (based on another issue, it seems this is fixed).

**Comment by noschang** [comment]
> Close due to staleness. Please reopen if the issue remains in the latest version (based on another issue, it seems this is fixed).

@jrandolf OK. Can you please post here the number of the issue that fixed the bug?

**Comment by jrandolf-google** [comment]
#359

**Comment by noschang** [comment]
> #359

@jrandolf Thank you very much!

**Comment by Petah** [comment]
I'm still getting this issue with `"puppeteer": "^18.0.4"`

Is there a work around that works? I have tried splitting the screenshot but that doesn't work either. 

![image](https://user-images.githubusercontent.com/134799/191638075-778ade7d-0fbb-4903-82f5-e9b81899d0b9.png)

**Comment by Sid-Turner-Ellis** [comment]
Im getting this too and on pretty small images

**Comment by raphaelApard** [comment]
Also still having this issue.

**Comment by sriraj66** [comment]
I solved the `ProtocolError: Unable to capture screenshot when taking screenshot of a large viewport` in Puppeteer by first checking the page's total scroll height using `document.documentElement.scrollHeight`, and then conditionally setting a safe maximum viewport height of 25,000px if the page exceeds that limit. 
Instead of using `fullPage: true` (which triggers the error on very tall pages), I captured a regular screenshot within this capped height, ensuring stability and avoiding Chrome’s internal rendering limits. For pages under the limit, you continued using `fullPage: true`, creating a dynamic and reliable screenshot solution for pages of any height.
```
import { launch } from "puppeteer";


(async () => {

    const browser = await launch({
      headless: true,
      devtools: false,
    });

    const page = await browser.newPage();
    await page.setViewport({
      width: 1920,
      height: 1080,
      deviceScaleFactor: 1,
    });

    const url = "https://dom.spec.whatwg.org/";
    await page.goto(url, {
      waitUntil: "networkidle2",
      timeout: 30000
    });
    const fullPageSize = await page.evaluate(() => {
        return {
          scrollWidth: document.documentElement.scrollWidth,
          scrollHeight: document.documentElement.scrollHeight,
        };
    });
    console.log(fullPageSize);

    try{

        if(fullPageSize.scrollHeight>25000){
            let original = page.viewport();
            await page.setViewport({
                width: original.width,
                height: 25000,
                deviceScaleFactor: original.deviceScaleFactor
            })
            await page.screenshot({
                path: "image.png"
            })
            await page.setViewport(original)

        }else{
            await page.screenshot({
                path: "image.png",
                fullPage: true
            })
        }
        
    }catch(err){
        console.log(err);
    }

    console.log(page.viewport());
    await browser.close()
  
})(); 
```

**Comment by AllesOderNicht** [comment]
Also still having this issue...
